tsflex/features/utils.py

```diff
@@ -2,13 +2,50 @@
 
 __author__ = "Jeroen Van Der Donckt, Jonas Van Der Donckt"
 
-import pandas as pd
+from typing import Callable, Any, Optional, List, Union, Tuple
+
 import numpy as np
+import pandas as pd
 
-from typing import Callable, Any, Optional, List, Union
 from .feature import FuncWrapper
 
 
+# ---------------------------------- PRIVATE METHODS ----------------------------------
+def _determine_bounds(bound_method, series_list: List[pd.Series]) -> Tuple[Any, Any]:
+    """Determine the bounds of the passed series.
+
+    Parameters
+    ----------
+    series_list : List[pd.Series]
+        The list of series for which the bounds are determined.
+
+    Returns
+    -------
+    Tuple[pd.Timestamp, pd.Timestamp]
+        The start & end timestamp, respectively.
+
+    """
+    if bound_method == "inner":
+        latest_start = series_list[0].index[0]
+        earliest_stop = series_list[0].index[-1]
+        for series in series_list[1:]:
+            latest_start = max(latest_start, series.index[0])
+            earliest_stop = min(earliest_stop, series.index[-1])
+        return latest_start, earliest_stop
+    elif bound_method == "outer":
+        earliest_start = series_list[0].index[0]
+        latest_stop = series_list[0].index[-1]
+        for series in series_list[1:]:
+            earliest_start = min(earliest_start, series.index[0])
+            latest_stop = max(latest_stop, series.index[-1])
+        return earliest_start, latest_stop
+    elif bound_method == "first":
```

Comment:
Does this make sense? In line 330 of `feature_collection.py` it is called on the values of a dict...

Reply:
What do you mean?

Reply:
Dict is not ordered, so calling values returns the values (i.e., series) in a 'random' order. Then it does not make sense to use bound_method = "first"

Reply:
Indeed, shouldn't be like that!

Reply:
Resolved in commit cf0928a